File: botstuff/dbconn.py
import json
import jsonpickle
from pymongo import MongoClient

db = None
config = {}

def conn():
    global db
    if db == None:
        client = MongoClient(config['host'])
        client.admin.authenticate(config['user'], config['pwd'], mechanism='SCRAM-SHA-1')
        uri = "mongodb://"+config['user']+":"+config['pwd']+"@"+config['host']+"/admin?authMechanism=SCRAM-SHA-1"
        db = MongoClient(uri).dueutil
        # client.drop_database('dueutil')
        return db
    else:
        return db

def insert_object(id,object):
    if id.strip() == "":
        return
    conn()[type(object).__name__].update({'_id':id},{"$set": {'data':jsonpickle.encode(object)}},upsert=True)

def get_collection_for_object(object_class):
    return conn()[object_class.__name__]

def load_config():
    global config
    with open('dbconfig.json') as config_file:  
        config = json.load(config_file)

load_config()

File: site/scripts/dbconn.php
<?php
require_once("util.php");


# Auth
$user = "admin";
$host = "127.0.0.1";
$pwd = "DueUtil4LyfeTest999AppleAndCheese";
$manager = new MongoDB\Driver\Manager("mongodb://$user:$pwd@$host/admin?authMechanism=SCRAM-SHA-1");
// $db = $client->selectDB("dueutil");

function get_collection_data($collection) {
    global $manager;
    
    // This is for collections made for this site that only have a single block a data
    $find_all_query = new MongoDB\Driver\Query(array());
    $cursor = $manager->executeQuery("dueutil.$collection",$find_all_query);
    $data = object_to_array($cursor->toArray()[0]);
    return $data;
}

function upsert($collection, $_id, $data, $set_mode='$set') {
    global $manager;
    $bulk = new MongoDB\Driver\BulkWrite;
    $bulk->update(['_id' => $_id], ['$set' => $data], ['upsert' => true]);
    $write_concern = new MongoDB\Driver\WriteConcern(MongoDB\Driver\WriteConcern::MAJORITY, 1000);
    $result = $manager->executeBulkWrite("dueutil.$collection", $bulk, $write_concern);
}
?>
